Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -0,0 +1,579 @@
+/*global define*/
+define([
+    '../ThirdParty/pako_inflate',
+    '../ThirdParty/when',
+    './appendForwardSlash',
+    './defaultValue',
+    './defined',
+    './defineProperties',
+    './DeveloperError',
+    './loadArrayBuffer',
+    './RuntimeError',
+    './TaskProcessor',
+    './throttleRequestByServer'
+], function(
+    pako,
+    when,
+    appendForwardSlash,
+    defaultValue,
+    defined,
+    defineProperties,
+    DeveloperError,
+    loadArrayBuffer,
+    RuntimeError,
+    TaskProcessor,
+    throttleRequestByServer) {
+    'use strict';
+
+    // Bitmask for checking tile properties
+    var childrenBitmasks = [0x01, 0x02, 0x04, 0x08];
+    var anyChildBitmask = 0x0F;
+    var cacheFlagBitmask = 0x10; // True if there is a child subtree
+    var imageBitmask = 0x40;
+    var terrainBitmask = 0x80;
+
+    function isBitSet(bits, mask) {
+        return ((bits & mask) !== 0);
+    }
+
+    /**
+     * Contains information about each tile from a Google Earth Enterprise server
+     *
+     * @param {Number} bits Bitmask that contains the type of data and available children for each tile.
+     * @param {Number} cnodeVersion Version of the request for subtree metadata.
+     * @param {Number} imageryVersion Version of the request for imagery tile.
+     * @param {Number} terrainVersion Version of the request for terrain tile.
+     *
+     * @private
+     */
+    function TileInformation(bits, cnodeVersion, imageryVersion, terrainVersion) {
+        this._bits = bits;
+        this.cnodeVersion = cnodeVersion;
+        this.imageryVersion = imageryVersion;
+        this.terrainVersion = terrainVersion;
+        this.ancestorHasTerrain = false; // Set it later once we find its parent
+        this.terrainState = 0; // UNKNOWN
```

Comment:
The state is unknown until we figure out whether the state is stored in this tile, in its parent or doesn't have any at all. There is an enum in the Terrain Provider that I use to set this state. It's only used within the terrain provider, but I can move it out to its own file if you think it would be better.